tests/test_send_message.py

```diff
@@ -116,3 +117,36 @@ def test_task_send_notification_catch_500(
 
     with pytest.raises(NotificationException):
         send_notification({"foo": "bar"})
+
+
+@freeze_time("2022-01-01")
+@pytest.mark.django_db(transaction=True)
+def test_api_create_person_unconfigured(api_client, notifications_config):
+    """Test the default behavior of NOTIFICATIONS_GUARANTEE_DELIVERY=True"""
+    notifications_config.delete()
+    url = reverse("person-list")
+    data = {"name": "John", "address_street": "Grotestraat", "address_number": "1"}
+
+    with patch(
+        "notifications_api_common.viewsets.send_notification.delay"
+    ) as mock_task:
+        with pytest.raises(RuntimeError):
+            response = api_client.post(url, data)
+
+    assert Person.objects.count() == 0
+
+
+@freeze_time("2022-01-01")
+@pytest.mark.django_db(transaction=True)
+@override_settings(NOTIFICATIONS_GUARANTEE_DELIVERY=False)
+def test_api_create_person_unconfigured(api_client, notifications_config):
+    notifications_config.delete()
+    url = reverse("person-list")
+    data = {"name": "John", "address_street": "Grotestraat", "address_number": "1"}
+
+    with patch(
+        "notifications_api_common.viewsets.send_notification.delay"
+    ) as mock_task:
```

Comment:
this mock should not be needed, so you can either remove it or add `mock_task.assert_not_called()` (and the same applies to the other test due to the bailing out via exception)